app/services/api/v3/database_validation/checks/path_positions_match_context_node_types.rb

```diff
@@ -0,0 +1,68 @@
+# Check if positions of nodes in flow paths match declared positions
+# in context_node_types.
+module Api
+  module V3
+    module DatabaseValidation
+      module Checks
+        class PathPositionsMatchContextNodeTypes < AbstractCheck
+          include ErrorMessageWithViolatingFlows
+
+          # Checks the flows table, expanding paths in a subjquery
+          # to get node types for each node and compares
+          # node_type_id + position from path versus that declared
+          # in context_node_types.
+          # @return (see AbstractCheck#passing?)
+          def passing?
+            @violating_flows = Api::V3::Flow.
+              select('a.id').
+              from(
+                <<~SQL
+                  (
+                    #{actual_column_positions_subquery.to_sql}
+                  ) a
+                  LEFT JOIN (
+                    #{expected_column_positions_subquery.to_sql}
+                  ) e ON a.position = (e.column_position + 1)
+                    AND a.node_type_id = e.node_type_id
+                SQL
+              ).where('e.node_type_id' => nil).
```

Comment:
Although, comparing on join is more performant. I don't know. A very simple test of comparing response time is telling me that comparing on join is better (a few seconds better for current sandbox database).

Reply:
So, it's ok.

Reply:
Lucky me, thanks for checking :) I'd think it should be faster, because the join should filter out a lot more records before applying the where, but I never checked!

Reply:
Yeah, exactly.